src/lib/UI/Module/ContentTree/NodeFactory.php

```diff
@@ -255,4 +222,89 @@ private function getSortClauses(
             return []; // rely on storage engine default sorting
         }
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo[] $contentInfoList
```

Comment:
Please provide an explanation here why the object reference is used instead of a normal array. At first glance it's not obvious that it **has** to be a reference because it's supposed to be secondary "return" that is used to load Content in bulk after the call.

Also possible rename the variable to better reflect what it's purpose is.

Reply:
Do you have any suggestions for the variable name? I thought that `$contentInfoList` is self-explanatory.

Reply:
These are the `ContentInfo`'s that have uninitialized `Content`, right?

`$uninitializedContentInfoList`?

You could also use `&array` instead of `ArrayObject` to denote the need to use an array reference here.

The thing is just `$contentInfoList` is not really correct because you only place `ContentInfo` objects there if their `Content` objects were not loaded if I'm correct, hence why I'd like to ask for that variable to be renamed.

Reply:
If so, the method declaration would be:
```php
private function buildNode(
        Location $location,
        array &$uninitializedContentInfoList,
        ?LoadSubtreeRequestNode $loadSubtreeRequestNode = null,
        bool $loadChildren = false,
        int $depth = 0,
        ?string $sortClause = null,
        string $sortOrder = Query::SORT_ASC
): Node {

Reply:
Ok, that's fine with me - thanks for explaining @Steveb-p :slightly_smiling_face: